File: app/src/main/java/com/stevenfrew/beatprompter/BeatEvent.java
package com.stevenfrew.beatprompter;

class BeatEvent extends BaseEvent
{
    double mBPM;
    int mBPB;
    int mBPL;
    int mBeat;
    boolean mClick;
    int mWillScrollOnBeat;

    BeatEvent(long eventTime,double bpm,int bpb,int bpl, int beat,boolean click,int willScrollOnBeat)
    {
        super(eventTime);
        mBPL=bpl;
        mPrevBeatEvent=this;
        mBPM=bpm;
        mBPB=bpb;
        mBeat=beat;
        mClick=click;
        mWillScrollOnBeat=willScrollOnBeat;
    }
}

File: app/src/main/kotlin/com/stevenfrew/beatprompter/song/event/BeatEvent.kt
package com.stevenfrew.beatprompter.song.event

/**
 * A BeatEvent signals the event processor to update the beat counter display, and optionally play a
 * metronome click sound (this will actually just generate a ClickEvent to represent the sound).
 */
class BeatEvent(
	eventTime: Long,
	val bpm: Double,
	var bpb: Int,
	val beat: Int,
	val click: Boolean,
	val willScrollOnBeat: Int
) : BaseEvent(eventTime) {
	override fun offset(nanoseconds: Long): BaseEvent =
		BeatEvent(eventTime + nanoseconds, bpm, bpb, beat, click, willScrollOnBeat)
}